controllers/chat/moveToSign.js

```diff
@@ -84,10 +84,10 @@ const moveToSign = async (req, res) => {
       })
     );
 
-    const oldChannelName = createdChannel?.channel?.name?.trim()?.split(',');
+    const oldChannelName = createdChannel?.channel?.name?.trim();
 
     try {
-      if (!oldChannelName) {
+      if (!oldChannelName || oldChannelName === ',') {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for generating a new channel name when `oldChannelName` is empty or only contains a comma seems fine. However, it's important to consider the case where `userModel?.username` or `targetUserModel?.username` are undefined or null. This could lead to an undesired channel name like "undefined,JohnDoe" or "JaneDoe,null". I suggest adding checks to ensure that both usernames exist before creating the new channel name.
```diff
- if (!oldChannelName || oldChannelName === ',') {
+ if ((!oldChannelName || oldChannelName === ',') && userModel?.username && targetUserModel?.username) {
```
If either username doesn't exist, you might want to handle this scenario differently, perhaps by throwing an error or assigning a default value.


<!-- This is an auto-generated comment by OSS CodeRabbit -->